File: Core/YABACore/src/commonMain/kotlin/dev/subfly/yabacore/state/detail/imagemark/ImagemarkDetailEvent.kt
package dev.subfly.yabacore.state.detail.imagemark

import dev.subfly.yabacore.notifications.PlatformNotificationText

sealed interface ImagemarkDetailEvent {
    data class OnInit(val bookmarkId: String) : ImagemarkDetailEvent
    data object OnDeleteBookmark : ImagemarkDetailEvent
    data object OnShareImage : ImagemarkDetailEvent
    data object OnExportImage : ImagemarkDetailEvent
    data object OnRequestNotificationPermission : ImagemarkDetailEvent
    data class OnScheduleReminder(
        val title: PlatformNotificationText,
        val message: PlatformNotificationText,
        val triggerAtEpochMillis: Long,
    ) : ImagemarkDetailEvent
    data object OnCancelReminder : ImagemarkDetailEvent
}

File: Darwin/YABACore/State/Detail/Imagemark/ImagemarkDetailEvent.swift
//
//  ImagemarkDetailEvent.swift
//  YABACore
//

import Foundation

public enum ImagemarkDetailEvent: Sendable {
    case onInit(bookmarkId: String)
    case onDeleteBookmark(bookmarkId: String)
    case onShareImage
    case onExportImage
    case onConsumePendingShare
    case onRequestNotificationPermission
    case onScheduleReminder(titleKey: String, messageKey: String, fireAt: Date)
    case onCancelReminder

    case updateSummary(bookmarkId: String, summary: String?)
}
